BUILD.md

```diff
@@ -0,0 +1,177 @@
+otherwise
+```shell
+$ cmake -DCMAKE_BUILD_TYPE=RELEASE . ../openMVG/src/
+```
+
+In order to use the MOSEK 6 back-end for the linear programming, OpenMVG checks that you have an up-to-date MOSEK licence, else OpenMVG MOSEK unit tests will fail.
+
+```shell
+$ cmake -DCMAKE_BUILD_TYPE=RELEASE
+    -DMOSEK_SEARCH_HEADER="~/Documents/Lib/mosek/6/tools/platform/linux64x86/h"
+    -DMOSEK_SEARCH_LIB="~/Documents/Lib/mosek/6/tools/platform/linux64x86/bin"
+    . ../openMVG/src/
+```
+
+If you want to have an IDE openable project with Code::Blocks:
+```shell
+$ cmake -G "CodeBlocks - Unix Makefiles" -DCMAKE_BUILD_TYPE=RELEASE . ../openMVG/src/
+```
+
+Compile the project
+```shell
+$ make
+```
+or for a multi-core compilation. (Replace NBcore with the number of threads)
+```shell
+$ make -j NBcore
+```
+
+Run tests (if requested at CMake step)
+```shell
+$ make test
+```
+
+Have fun with the samples
+```shell
+$ cd openMVG_Samples
+```
+
+Windows compilation
+-------------------
+
+Checkout the project
+```shell
+$ git clone --recursive https://github.com/openMVG/openMVG.git
+```
+
+1. Open cmake-gui.
+2. Fill the source path with the src OpenMVG path.
+3. Fill the build path with a new directory.
+4. Select your Visual Studio IDE and click configure and then generate.
+5. Open the .sln solution created in your build directory.
+6. Change the target to Release.
+7. Compile the libraries and binaries samples.
+
+Mac compilation
+-------------------
+
+```shell
+$ git clone --recursive https://github.com/openMVG/openMVG.git
+$ cd openMVG
+$ ls
+ AUTHORS BUILD  docs  logo  README  src  ...
+$ cd ..
+$ mkdir openMVG_Build
+$ cd openMVG_Build
+$ cmake -DCMAKE_BUILD_TYPE=RELEASE -G "Xcode" . ../openMVG/src/
+```
+If you want to add unit tests and examples to the build, run
+```shell
+$ cmake -DCMAKE_BUILD_TYPE=RELEASE -DOpenMVG_BUILD_TESTS=ON -DOpenMVG_BUILD_EXAMPLES=ON -G "Xcode" . ../openMVG/src/
+```
+otherwise
+```shell
+$ cmake -DCMAKE_BUILD_TYPE=RELEASE -G "Xcode" . ../openMVG/src/
+```
+```shell
+$ xcodebuild -configuration Release
+```
+
+Using OpenCV sample
+--------------------
+
+Add `-DOpenMVG_USE_OPENCV=ON` to your cmake command and set the OpenCV_DIR variable to your OpenCV build directory
+e.g. `-DOpenCV_DIR="/home/user/Dev/github/itseez/opencv_Build" -DOpenMVG_USE_OPENCV=ON`
+
+Using OpenMVG as a third party library dependency in CMake
+-------------------------------------------------------------
+
+OpenMVG can be used as a third party library once it has been installed.
+Because it can use its own Ceres version, it's better to install it locally and not in system files.
+So please consider using the CMAKE_INSTALL_PREFIX CMake variable to specify a local installation directory.
+
+Here is the syntax to add the variable to the cmake command (use absolute path):
+`-DCMAKE_INSTALL_PREFIX:STRING="YourInstallPath"`
+e.g `-DCMAKE_INSTALL_PREFIX:STRING="/home/user/Dev/github/openMVG_Build/openMVG_install"`
+
+Perform `make` and `make install`.
```

Comment:
Maybe this is not true for IDE builds ? (IDE should have an INSTALL target that have to be executed to perform the installation).